pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -231,6 +231,7 @@ type SpreadConstraint struct {
 	SpreadByLabel string `json:"spreadByLabel,omitempty"`
 
 	// MaxGroups restricts the maximum number of cluster groups to be selected.
+	// Defaults to MinGroups.
```

Comment:
I don't think the `MaxGroups` deserves a default value.
In general, the default value should be a common one that meets most of the use cases.
Here, the valid range of `MinGroups` is [1, max(int)], in order to let the scheduler works out(or, try to) a result we set the defaults to `1`, it seems reasonable, but for `MaxGroups` which is essentially a limitation to the scheduler that can't spread the group that much, we don't have enough evidence to show that users always set `MaxGroups` with the same value as `MinGroups`.

Reply:
@RainbowMango Hi, thanks for your replies!
I agree with most of your thoughts, but `MaxGroups` would be zero which is an incomprehensible value for ` karmada-scheduler` without a default value. In other words, how should we(or ` karmada-scheduler` does) understand `MaxGroups` without a default value?

Reply:
> how should we(or  karmada-scheduler does) understand MaxGroups without a default value?

It perhaps can't happen because there is a validation to ensure `MaxGroups` can't be lower than `MinGroups`(1).
right?